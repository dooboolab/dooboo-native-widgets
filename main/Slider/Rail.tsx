import React, { FC } from 'react';
import { StyleProp, ViewStyle } from 'react-native';

import styled from 'styled-components/native';

interface Props {
  testID?: string;
  style?: StyleProp<ViewStyle>;
}

const DEFAULT = {
  height: 3,
  borderRadius: 1,
};

const StyledRail = styled.View`
  width: 100%;
  height: ${`${DEFAULT.height}px`};
  border-radius: ${DEFAULT.borderRadius}px;
  background-color: #bcdbfb;
`;

const Rail: FC<Props> = ({ testID, style }) => {
  return <StyledRail testID={testID} style={style} />;
};

export default Rail;
