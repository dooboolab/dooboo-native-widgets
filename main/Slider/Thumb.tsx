import { Animated, StyleProp, ViewStyle } from 'react-native';
import React, { FC } from 'react';

import styled from 'styled-components/native';

interface ThumbPositionerType {
  percent: number;
}

const ThumbPositioner = styled.View<ThumbPositionerType>`
  position: absolute;
  left: ${({ percent }): string => `${percent}%`};
`;

const StyledThumb = styled.View`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 12px;
  height: 12px;
  background-color: #0b21e8;
  border-radius: 6px;
  transform: translate(-6px);
`;

interface Props {
  testID?: string;
  percent: number;
  size?: number;
  scaleValue?: Animated.Value;
  opacityValue?: Animated.Value;
  customThumb?: React.ReactElement;
  style?: StyleProp<ViewStyle>;
}

const Thumb: FC<Props> = ({
  testID,
  percent,
  size = 12,
  scaleValue = new Animated.Value(0.01),
  opacityValue = new Animated.Value(0.12),
  customThumb,
  style,
}) => {
  const rippleSize = (size - 2) * 2 + 48 / size;

  return (
    <ThumbPositioner testID="thumb-positioner-test-id" percent={percent}>
      {customThumb || (
        <StyledThumb testID={testID} style={style}>
          <Animated.View
            style={{
              width: rippleSize,
              height: rippleSize,
              borderRadius: 100,
              transform: [{ scale: scaleValue }],
              opacity: opacityValue,
              backgroundColor: '#0b21e8',
            }}
          />
        </StyledThumb>
      )}
    </ThumbPositioner>
  );
};

export default Thumb;
