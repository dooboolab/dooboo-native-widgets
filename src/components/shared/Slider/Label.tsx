import { Animated } from 'react-native';
import React from 'react';
import styled from 'styled-components/native';

interface Props {
  percentValue: Animated.Value;
  value: number;
  size?: number;
  testID?: string;
}

const Value = styled.Text`
  transform: rotate(45deg);
  color: white;
`;

const Label: React.FC<Props> = ({ percentValue, value, size = 32, testID }) => {
  return (
    <Animated.View
      testID={testID}
      style={{
        position: 'absolute',
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        width: size,
        height: size,
        borderTopRightRadius: size / 2,
        borderTopLeftRadius: size / 2,
        borderBottomRightRadius: size / 2,
        backgroundColor: '#4163f4',
        transform: [
          { rotate: '-45deg' },
          { translateX: -((Math.sqrt(2) * size) / 2) },
        ],
        top: -size * 2,
        left: percentValue.interpolate({
          inputRange: [0, 100],
          outputRange: ['0%', '100%'],
        }),
      }}
    >
      <Value>{value}</Value>
    </Animated.View>
  );
};

export default Label;
