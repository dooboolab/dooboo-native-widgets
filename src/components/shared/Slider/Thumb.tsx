import {
  Animated,
  Easing,
  Platform,
  TouchableWithoutFeedback,
} from 'react-native';
import React, { FC, ReactNode, useState } from 'react';
import styled from 'styled-components/native';

const Container = styled.View<{ percent: number }>`
  position: absolute;
  width: 12;
  height: 12;
  background-color: #0b21e8;
  border-radius: 6;
  top: -4.5;
  transform: translate(-6px);
  left: ${({ percent }): string => `${percent}%`};
`;

interface Props {
  size?: number;
  percent: number;
  defaultPercent?: number;
}

const Thumb: FC<Props> = ({ size = 12, percent }) => {
  const [scaleValue, setScaleValue] = useState(new Animated.Value(0.01));
  const [opacityValue, setOpacityValue] = useState(new Animated.Value(0.12));
  const onPressedIn = (): void => {
    Animated.timing(scaleValue, {
      toValue: 1,
      duration: 225,
      easing: Easing.bezier(0.0, 0.0, 0.2, 1),
      useNativeDriver: Platform.OS === 'android',
    }).start();
  };

  const onPressedOut = (): void => {
    Animated.timing(opacityValue, {
      toValue: 0,
    }).start(() => {
      setOpacityValue(new Animated.Value(0.12));
      setScaleValue(new Animated.Value(0.01));
    });
  };

  const rippleSize = size * 2;
  const renderRippleView = (): ReactNode => {
    return (
      <Animated.View
        style={{
          position: 'absolute',
          top: -6,
          left: -6,
          width: rippleSize,
          height: rippleSize,
          borderRadius: rippleSize / 2,
          transform: [{ scale: scaleValue }],
          opacity: opacityValue,
          backgroundColor: 'black',
        }}
      />
    );
  };

  return (
    <TouchableWithoutFeedback onPressIn={onPressedIn} onPressOut={onPressedOut}>
      <Container percent={percent}>
        {renderRippleView()}
      </Container>
    </TouchableWithoutFeedback>
  );
};

export default Thumb;
