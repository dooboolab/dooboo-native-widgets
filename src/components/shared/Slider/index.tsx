import * as React from 'react';

import styled from 'styled-components/native';

export interface SliderProps {
}

function SearchInput(props: SliderProps): React.ReactElement {
  return (
  );
}