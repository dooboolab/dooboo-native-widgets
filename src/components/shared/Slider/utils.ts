export const valueToPercent = (value, maxValue, minValue): number => {
  return (value / (maxValue - minValue)) * 100;
};

export const percentToValue = (percent, maxValue, minValue): number => {
  return ((maxValue - minValue) * percent) / 100;
};

export const getPercent = (positionX: number, sliderWidth: number): number => {
  const percent = (positionX / sliderWidth) * 100;
  if (percent <= 0) return 0;
  else if (percent >= 100) return 100;
  else return percent;
};
