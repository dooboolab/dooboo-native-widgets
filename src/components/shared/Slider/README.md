# Slider

[![Npm Version](http://img.shields.io/npm/v/@dooboo-ui/native-slider.svg?style=flat-square)](https://npmjs.org/package/@dooboo-ui/native-slider)
[![Downloads](http://img.shields.io/npm/dm/@dooboo-ui/native-slider.svg?style=flat-square)](https://npmjs.org/package/@dooboo-ui/native-slider)

<img src="https://user-images.githubusercontent.com/11019960/73190141-dee25300-4168-11ea-8ba0-f806c2307f9a.png" width="400" />

## Props

|                      | necessary | types                   | default |
| -------------------- | --------- | ----------------------- | ------- |
| hideMark             |           | boolean                 |false    |
| hideLabel            |           | boolean                 |true     |
| autoLabel            |           | boolean                 |false    |
| step                 |           | number                  |1        |
| defaultValue         |           | number                  |0        |
| minValue             |           | number                  |0        |
| maxValue             |           | number                  |100      |
| onChange             |           | (value: number) => void |-        |

## Installation

```sh
yarn add @dooboo-ui/native
```

or

```sh
yarn add @dooboo-ui/native-slider
```

## Usage

```javascript
import React, { ReactElement } from 'react';

import { Slider } from '@dooboo-ui/native';
// or
import Slider from '@dooboo-ui/native-slider';
import styled from 'styled-components/native';

function Page(): ReactElement {
  const handleChange = (value) => {
    // Do something!
  };

  return (
    <Container>
      <Slider
        step={10}
        defaultValue={20}
        minValue={0}
        maxValue={100}
      />

      <Slider
        minValue={0}
        maxValue={5}
        step={1}
        hideLabel={false}
        autoLabel
        onChange={handleChange}
      />
    </Container>
  );
}

export default Page;
```

