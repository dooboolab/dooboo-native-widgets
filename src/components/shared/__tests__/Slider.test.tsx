import * as React from 'react';

import { Animated, View } from 'react-native';
import { fireEvent, render } from '@testing-library/react-native';
import {
  getNearestPercentByValue,
  getPercentByPositionX,
  getPercentByValue,
  getStepValueByPercent,
  getValueByPercent,
  roundNearest,
} from '../Slider/utils';

import Label from '../Slider/Label';
import Marks from '../Slider/Marks';
import Rail from '../Slider/Rail';
import Slider from '../Slider';
import Thumb from '../Slider/Thumb';
import Track from '../Slider/Track';

const TEST_ID = {
  RAIL: 'rail-test-id',
  TRACK: 'track-test-id',
  MARKS: 'marks-test-id',
  THUMB: 'thumb-test-id',
  LABEL: 'label-test-id',
  THUMBPOSITIONER: 'thumb-positioner-test-id',
};

describe('[Slider] render', () => {
  it('renders without crashing', () => {
    const rendered = render(
      <Slider
        hideMark={false}
        defaultValue={10}
        minValue={0}
        maxValue={100}
        step={10}
        onChange={(): void => {}}
      />,
    ).asJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  describe('required components', () => {
    it('should have a [Rail].', () => {
      const { queryByTestId } = render(
        <Slider />,
      );
      const rail = queryByTestId(TEST_ID.RAIL);

      expect(rail).not.toBeNull();
    });

    it('should have a [Track].', () => {
      const { queryByTestId } = render(
        <Slider />,
      );
      const track = queryByTestId(TEST_ID.TRACK);

      expect(track).not.toBeNull();
    });

    it('should have a [Marks].', () => {
      const { queryByTestId } = render(
        <Slider />,
      );
      const marks = queryByTestId(TEST_ID.MARKS);

      expect(marks).not.toBeNull();
    });

    it('should have a [Thumb].', () => {
      const { queryByTestId } = render(
        <Slider />,
      );
      const thumb = queryByTestId(TEST_ID.THUMB);

      expect(thumb).not.toBeNull();
    });

    it('should hide [Label].', () => {
      const { queryByTestId } = render(
        <Slider />,
      );
      const label = queryByTestId(TEST_ID.LABEL);

      expect(label).toBeNull();
    });

    it('should have a [Label].', () => {
      const { queryByTestId } = render(
        <Slider
          hideLabel={false}
        />,
      );
      const label = queryByTestId(TEST_ID.LABEL);

      expect(label).not.toBeNull();
    });
  });

  it('should hide [Marks] when hideMark is true.', () => {
    const { queryByTestId } = render(
      <Slider
        hideMark={true}
      />,
    );
    const marks = queryByTestId(TEST_ID.MARKS);

    expect(marks).toBeNull();
  });

  it('should hide [Marks] when step is less than 0.', () => {
    const { queryByTestId } = render(
      <Slider
        step={-1}
      />,
    );
    const marks = queryByTestId(TEST_ID.MARKS);

    expect(marks).toBeNull();
  });

  it('should hide [Marks] when step is equal to 0.', () => {
    const { queryByTestId } = render(
      <Slider
        step={0}
      />,
    );
    const marks = queryByTestId(TEST_ID.MARKS);

    expect(marks).toBeNull();
  });
});

describe('[Marks]', () => {
  const Mark: React.FC<{ testID: string }> = ({ testID }) => {
    return <View testID={testID} />;
  };

  it('renders without crashing', () => {
    const rendered = render(
      <Marks
        sliderWidth={100}
        mark={<Mark testID={TEST_ID.MARKS} />}
        customMarkWidth={1}
        step={10}
        minValue={0}
        maxValue={100}
      />,
    ).asJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  it('calls onMarkPress with expected arguments', () => {
    const handleMarkPress = jest.fn();
    const STEP = 10;
    const MARK_COUNT = 10;

    const { getAllByTestId } = render(
      <Marks
        sliderWidth={100}
        mark={<Mark testID={TEST_ID.MARKS} />}
        customMarkWidth={10}
        step={STEP}
        minValue={0}
        maxValue={100}
        onMarkPress={handleMarkPress}
      />,
    );
    const marks = getAllByTestId(TEST_ID.MARKS);
    for (let i = 0; i < MARK_COUNT; i += 1) {
      fireEvent.press(marks[i]);
    }
    for (let i = 0; i < MARK_COUNT; i += 1) {
      const [value, position, index] = handleMarkPress.mock.calls[i];
      expect(value).toBe(i * STEP);
      expect(index).toBe(i);
    }
  });

  it('calls onInit with expected arguments', () => {
    const onInit = jest.fn();
    const STEP = 8;
    const MIN_VALUE = 0;
    const MAX_VALUE = 96;
    const MARK_COUNT = Math.floor((MAX_VALUE - MIN_VALUE) / STEP);

    render(
      <Marks
        sliderWidth={100}
        mark={<Mark testID={TEST_ID.MARKS} />}
        customMarkWidth={12}
        style={{ width: 240 }}
        step={STEP}
        minValue={MIN_VALUE}
        maxValue={MAX_VALUE}
        onInit={onInit}
      />,
    );
    expect(onInit).toHaveBeenCalledTimes(1);
    const expectedArgumentForOnInit = Array.from({
      length: MARK_COUNT + 1,
    }).map((_, idx) => STEP * idx);
    const actualArgumentForOnInit = onInit.mock.calls[0][0];

    expect(actualArgumentForOnInit).toStrictEqual(expectedArgumentForOnInit);
  });
});

describe('[Rail]', () => {
  it('renders without crashing', () => {
    const rendered = render(
      <Rail testID="RailTestID" style={{ backgroundColor: 'red' }} />,
    ).asJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});

describe('[Label]', () => {
  it('renders without crashing', () => {
    const rendered = render(
      <Label
        percentValue={new Animated.Value(0)}
        testID="LabelTestID"
        value={0}
      />,
    ).asJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});

describe('[Track]', () => {
  it('renders without crashing', () => {
    const rendered = render(
      <Track
        testID="TrackTestID"
        style={{ backgroundColor: 'red' }}
        percent={50}
      />,
    ).asJSON();
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });

  it('should be n% width of Slider.', () => {
    const TEST_ID_TRACK = 'TrackTestID';
    const percent = 50;
    const { getByTestId } = render(
      <Track
        testID="TrackTestID"
        style={{ backgroundColor: 'red' }}
        percent={50}
      />,
    );
    const track = getByTestId(TEST_ID_TRACK);

    expect(track.getProp('width')).toBe(percent);
  });

  describe('[Track]', () => {
    it('renders without crashing', () => {
      const rendered = render(
        <Track
          testID="TrackTestID"
          style={{ backgroundColor: 'red' }}
          percent={50}
        />,
      ).asJSON();
      expect(rendered).toMatchSnapshot();
      expect(rendered).toBeTruthy();
    });

    it('should be n% width of Slider.', () => {
      const TEST_ID_TRACK = 'TrackTestID';
      const percent = 50;
      const { getByTestId } = render(
        <Track
          testID={TEST_ID_TRACK}
          style={{ backgroundColor: 'red' }}
          percent={50}
        />,
      );
      const track = getByTestId(TEST_ID_TRACK);

      expect(track.getProp('width')).toBe(percent);
    });
  });

  describe('[Thumb]', () => {
    it('renders without crashing', () => {
      const rendered = render(
        <Thumb
          testID="ThumbTestID"
          opacityValue={new Animated.Value(0.12)}
          scaleValue={new Animated.Value(0.01)}
          percent={50}
          size={10}
        />,
      ).asJSON();
      expect(rendered).toMatchSnapshot();
      expect(rendered).toBeTruthy();
    });

    it('should pose [Thumb] to given percent.', () => {
      const PERCENT = 30;
      const { container } = render(
        <Thumb
          percent={PERCENT}
        />,
      );
      const thumb = container.props.children;

      expect(thumb.props.percent).toBe(PERCENT);
    });
  });

  describe('utils', () => {
    describe('roundNearest', () => {
      it('should return floored when the number is smaller than half of the digit.', () => {
        const result = roundNearest(12, 5);

        expect(result).toBe(10);
      });
      it('should return ceiled when the number is smaller than half of the digit.', () => {
        const result = roundNearest(13, 5);

        expect(result).toBe(15);
      });
    });

    describe('getPercentByValue', () => {
      it('should return the rate of value between minValue and maxValue.', () => {
        const result = getPercentByValue(5, 60, 10);

        expect(result).toBe(10);
      });
    });

    describe('getValueByPercent', () => {
      it('should return the value of rate between minValue and maxValue.', () => {
        const result = getValueByPercent(10, 60, 10);

        expect(result).toBe(5);
      });
    });

    describe('getPercentByPositionX', () => {
      it('should return 0 when the rate of positionX is equal or smaller than 0.', () => {
        const result = getPercentByPositionX({
          positionX: -1,
          sliderWidth: 100,
          stepPercent: 5,
        });

        expect(result).toBe(0);
      });

      it('should return 100 when the rate of positionX is equal or smaller than 100.', () => {
        const result = getPercentByPositionX({
          positionX: 110,
          sliderWidth: 100,
          stepPercent: 5,
        });

        expect(result).toBe(100);
      });

      it('should return the floored rate of positionX according to step.', () => {
        const result = getPercentByPositionX({
          positionX: 12,
          sliderWidth: 100,
          stepPercent: 5,
        });

        expect(result).toBe(10);
      });

      it('should return the ceiled rate of positionX according to step.', () => {
        const result = getPercentByPositionX({
          positionX: 13,
          sliderWidth: 100,
          stepPercent: 5,
        });

        expect(result).toBe(15);
      });
    });

    describe('getNearestPercentByValue', () => {
      it('should return 0 when the rate of value is equal or smaller than 0.', () => {
        const result = getNearestPercentByValue({
          value: 1,
          minValue: 10,
          maxValue: 60,
          step: 5,
        });

        expect(result).toBe(0);
      });

      it('should return 100 when the rate of value is equal or smaller than 100.', () => {
        const result = getNearestPercentByValue({
          value: 70,
          minValue: 10,
          maxValue: 60,
          step: 5,
        });

        expect(result).toBe(100);
      });

      it('should return the floored rate according to value between minValue and maxValue.', () => {
        const result = getNearestPercentByValue({
          value: 12,
          minValue: 10,
          maxValue: 60,
          step: 5,
        });

        expect(result).toBe(20);
      });

      it('should return the ceiled rate according to value between minValue and maxValue.', () => {
        const result = getNearestPercentByValue({
          value: 13,
          minValue: 10,
          maxValue: 60,
          step: 5,
        });

        expect(result).toBe(30);
      });
    });

    describe('getStepValueByPercent', () => {
      it('should return value of step according to percent.', () => {
        const result = getStepValueByPercent({
          percent: 10,
          stepPercent: 40,
          step: 5,
        });

        expect(result).toBe(1.25);
      });
    });
  });
});
