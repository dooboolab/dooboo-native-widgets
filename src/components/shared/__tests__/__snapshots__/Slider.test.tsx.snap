// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Slider] render renders without crashing 1`] = `
<View
  onMoveShouldSetResponder={[Function]}
  onMoveShouldSetResponderCapture={[Function]}
  onResponderEnd={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderReject={[Function]}
  onResponderRelease={[Function]}
  onResponderStart={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onStartShouldSetResponderCapture={[Function]}
  style={
    Array [
      Object {},
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "display": "flex",
          "justifyContent": "center",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#90A4F9",
            "borderRadius": 1,
            "height": 3,
            "width": 250,
          },
          Object {},
        ]
      }
      theme={
        Object {
          "backgroundColor": "#90A4F9",
        }
      }
    />
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={-8.801282051282051}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": -8.801282051282051,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={11.801282051282051}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 11.801282051282051,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={32.40384615384615}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 32.40384615384615,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={53.006410256410255}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 53.006410256410255,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={73.60897435897436}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 73.60897435897436,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={94.21153846153845}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 94.21153846153845,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={114.81410256410257}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 114.81410256410257,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={135.41666666666669}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 135.41666666666669,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={156.01923076923077}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 156.01923076923077,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={176.62179487179486}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 176.62179487179486,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={197.22435897435898}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 197.22435897435898,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={217.8269230769231}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 217.8269230769231,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
    <View
      accessible={true}
      clickable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      position={238.42948717948718}
      style={
        Array [
          Object {
            "alignItems": "center",
            "display": "flex",
            "left": 238.42948717948718,
            "position": "absolute",
            "width": 20.602564102564102,
          },
        ]
      }
      width={20.602564102564102}
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#4163F4",
              "height": 3,
              "width": 3,
            },
            Object {},
          ]
        }
        theme={
          Object {
            "backgroundColor": "#4163F4",
          }
        }
      />
    </View>
  </View>
</View>
`;
